pkg/crc/services/dns/dns_darwin.go

```diff
@@ -123,37 +125,37 @@ func waitForNetwork(hostResolv *network.ResolvFileValues) error {
 	return fmt.Errorf("Host is not connected to internet.")
 }
 
-// updateResolvConfFile updates the host's /etc/resolv.conf file with Instance IP.
-func updateResolvConfFile(instanceIP string, resolvConfFile string) (bool, error) {
-	// Get the current value of resolv.conf file
-	hostResolv, err := network.GetResolvValuesFromHost()
+// updateHostsConfFile updates the host's /etc/hosts file with Instance IP.
+func updateHostsConfFile(instanceIP string) (bool, error) {
+	// Get the current value of /etc/hosts file
+	hosts, err := goodhosts.NewHosts()
 	if err != nil {
-		return false, fmt.Errorf("Unable to read host resolv file (%v)", err)
+		return false, err
 	}
-
-	foundExistingInstanceIP := false
-
-	for i, ns := range hostResolv.NameServers {
-		// Update the nameserver IP with instance IP if already exist
-		if strings.Contains(ns.IPAddress, "192.168.130") {
-			hostResolv.NameServers[i].IPAddress = instanceIP
-			foundExistingInstanceIP = true
-			break
+	// Check if the Domain already present if yes then remove those
+	if hosts.HasHostname(constants.DefaultAPIDomain) {
```

Comment:
`constants.DefaultAPIDomain` can be used if a fallback is needed, but the value should be built from `baseDomain`, `clusterName` and `appsDomain` from `crc-bundle-info.json`